go/hamming: disapprove comments (#1224)
exercism/website-copy: automated-comments/go/hamming/comparing_bytes_disguised_as_runes.md

> Converting the `bytes` to `runes` to compare disguises the problem.

Reviewer: What problem?

Author: Rephrased

> It works perfectly in the context of this exercise as there are only ascii letters in DNA strands.

Reviewer 2: I think we would usually write 'ASCII', not 'ascii'—it's an acronym.

Reviewer: ```suggestion
It works perfectly in the context of this exercise as there are only ASCII letters in DNA strands.
```

> It works perfectly in the context of this exercise as there are only ascii letters in DNA strands.

Reviewer: ```suggestion
Such conversion works perfectly in the context of this exercise as there are only ASCII letters in DNA strands.
```

Grammarly, `it` refers to "the problem"

> Try creating more tests with non-ascii characters and see what happens.

Reviewer 2: I think this is likely to be a bit beyond the capabilities of students so early in the track, and unless they do that, they won't get the point. Do we need a more detailed explanation of the problem here?

I'm not completely clear what triggers this comment, but I'm guessing it's something like indexing the string by byte, and then converting that byte to a rune before comparing it. The right thing to do (I usually propose) is to convert both strings to rune sequences _first_, before doing anything with them. Then it's meaningful to compare their lengths, and safe to iterate over them comparing element with element.

Reviewer: ```suggestion
Try creating more tests with non-ASCII characters and see what happens.
```

(same fix as above).

For the rest I agree with @bitfield .

Author: Yes, this is triggered when doing something like this:

```go
for i, r := range a {
    if r == rune(b[i])
}
```

I'll look into explaining the "problem" better. I think I would stick with letting the user find out what to do about it as this is more of a challenge rather than a "must". That should probably be clearer as well.

Converting a `byte` to a `rune` works perfectly in the context of this exercise as there are only ASCII letters in DNA strands.

> ASCII characters consist of one `byte` each. As soon as non-ASCII characters get involved this will break as they consist of 

Reviewer: ```suggestion
ASCII characters consist of one `byte` each. As soon as non-ASCII characters get involved this method of comparison breaks as those characters consist of 
```

Just solidifying those references, as this sorta refers back to "this exercise" when "this method of comparing" (converting a byte to a rune and then doing the comparison) is meant.

> up to 4 bytes.

Reviewer 3: Should this line be combined with the previous one? (Not sure how whitespace is treated in the UI)

Reviewer: New lines are mark-down new lines, so a treated as a space if there's only one.
